generator/internal/dart/annotate.go

```diff
@@ -121,22 +122,40 @@ type packageDependency struct {
 	Constraint string
 }
 
+type AnnotateModel struct {
+	// The API model we're annotating.
+	model *api.API
+	// Mappings from IDs to types.
+	state *api.APIState
+	// The set of imports that have been calculated.
+	imports map[string]string
+	// The mapping from protobuf packages to Dart import statements.
+	packageMapping map[string]string
+	// A mapping from field IDs to fields for the fields we know to be required.
+	requiredFields map[string]*api.Field
+}
+
+func NewAnnotateModel(model *api.API) *AnnotateModel {
```

Comment:
ditto: prefer private visibility for most functions and and types unless you really intend to use them from outside the package.